### DISCUSSION TITLE
Nesting scss and BEM naming in built-in sass support

### DISCUSSION THREAD

**Discussion by alikleitcr7** [vercel/next.js#27772] [Help]
How to properly use scss following BEM naming conversion in built-in sass module?

Consider the following:

jsx:
```jsx
<div className="filter-items">
   <div className="filter-item">
     <div className="filter-item__title">
           <span><span><span></span>
     </div>
   <div>
<div>
```

with scss:
```scss
.filter-items{
    .filter-item{
         &__title { 
              span:first-child{}
         }
    }
}
```

when I console log this styles object after importing the css module I get `['{path}-filter-items-{somecode}','{path}-filter-item-{somecode}']`

In this stylesheet, the title `__title` won't take effect, though it is not presented in the array as well, is this naming convention not included in the build-in support?

**Comment by morganfeeney** [comment]
Are you using built-in CSS support A.K.A. CSS Modules: https://nextjs.org/docs/basic-features/built-in-css-support#sass-support? If you are, you will need to import the styles first and access them using dot notation, e.g.

```jsx
import style from 'yourstyles.module.scss';
...
<div className={style.filterItems}>
...
<div>
```

**Reply by alikleitcr7** [reply]
I did, but it won't take effect for `__title` . installed sass version `1.32.8`. 

and since my classes are kebab-case. I'm accessing through array notation `styles['filter-items']`

**Reply by morganfeeney** [reply]
Why don't you change to camelCase? You'll get tired of using bracket notation eventually.

Post some code samples please, so I can see what you are trying to do to access `__title`.

**Reply by morganfeeney** [reply]
I just setup a reduced test-case using `create-next-app` using the code supplied:

```scss
.filter-items{
  .filter-item{
    &__title {
      font-size: 3rem;
      span:first-child{
        background: red}
    }
  }
}
```
```jsx
<div className={scss["filter-items"]}>
  <div className={scss["filter-item"]}>
    <div className={scss["filter-item__title"]}>
      <span>test 1</span><span>test 2</span>
    </div>
  </div>
</div>
```
Output to the console the style object containing some selectors, note that there is no pseudo class.
```js
{
    "filter-items": "sassy_filter-items__2cSq4",
    "filter-item__title": "sassy_filter-item__title__NV7UB"
}
```
See screenshots for effects:
![image](https://user-images.githubusercontent.com/1900222/128593864-6a1efca2-c770-4ad4-b89e-a717cc83f38f.png)

![image](https://user-images.githubusercontent.com/1900222/128593755-bf36bf07-4f55-47e3-863c-e4999aa43ad2.png)

I'm not seeing any issues with how Nextjs supports this syntax.

```json
{
  "name": "sassy",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "next": "11.0.1",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "sass": "^1.37.5"
  },
  "devDependencies": {
    "eslint": "7.32.0",
    "eslint-config-next": "11.0.1"
  }
}

```

I also tried downgrading sass to 1.32.8 and had no issues

**Reply by alikleitcr7** [reply]
Thanks 👍 ! It then worked for me after placing at least one style in the class `__title`. Interesting thing is that **the imported style object won't contain the class if the class has no styles**.

**Reply by morganfeeney** [reply]
It works without adding a style to the class, even with only a pseudo on the span.
